dataprofiler/tests/profilers/test_json_encoder.py

```diff
@@ -29,3 +31,60 @@ def test_encode_base_column_profiler(self):
         )
 
         self.assertEqual(json.loads(serialized), exepcted)
+
+    def test_encode_categorical_column_profiler(self):
+        profile = CategoricalColumn("0")
+
+        serialized = json.dumps(profile, cls=ProfileEncoder)
+        expected = json.dumps(
+            {
+                "name": "0",
+                "col_index": np.nan,
+                "sample_size": 0,
+                "metadata": dict(),
+                "times": defaultdict(),
+                "thread_safe": True,
+                "_categories": defaultdict(int),
+                "_CategoricalColumn__calculations": dict(),
+                "_top_k_categories": None,
+            }
+        )
+
+        self.assertEqual(json.loads(serialized), json.loads(expected))
```

Comment:
if we are trying to test the functionality of encoding and decoding (profile -> json -> profile) then I think that the assertEqual() should be comparing the decoded profile to another profile object that represents the object we expect as a result.